common/goos/Object.cpp

```diff
@@ -83,6 +83,31 @@ std::string object_type_to_string(ObjectType type) {
   }
 }
 
+// converts an object type to it's symbol name to be converted into a typespec
+// returns 'object' otherwise
+std::string object_type_to_type_name(ObjectType kind) {
```

Comment:
do you mind moving this function to somewhere like CompilerControl.cpp? I think this is confusing to have in the general goos folder because this is sort of converting between goos and goal types.